lxd/images.go

```diff
@@ -1005,6 +1018,18 @@ func autoUpdateImage(d *Daemon, op *operation, id int, info *api.Image) error {
 	return nil
 }
 
+func pruneExpiredImagesTask(d *Daemon) (task.Func, task.Schedule) {
+	f := func(context.Context) {
+		pruneExpiredImages(d)
+	}
+
+	// Skip the first run, and instead run an initial pruning synchronously
+	// before we start updating images later on in the start up process.
+	pruneExpiredImages(d)
```

Comment:
Feels like some of the code from pruneExpiredImages should actually be moved here to make it more consistent with the image updating code above.

Reply:
I'm not sure what you mean here. In the image update code above we have 1) autoUpdateImagesTask which holds the logic for deciding when images should be updated 2) autoUpdateImages that actually updates images. The pruneExpiredImagesTask and pruneExpiredImages pretty much match that pattern.

Reply:
My point is that rather than just always running this once a day, the schedule should be set based on the value in images.remote_cache_expiry.

Right now if you look at autoUpdateImagesTask and pruneExpiredImagesTask, you'll notice that the former does the config + scheduling check directly through the tasks package whereas the latter will always call the function once a day and it's then up to the function in lxd/images.go to read the config and skip execution.

Reply:
I see. I've changed it like you suggest in bce0b923271f02f5326135926441752fa1ed4c0a, but note that the code got more verbose an less DRY (due to the repeated checks on daemonConfig["images.remote_cache_expiry"]). Not sure it's really worth, but your take.